src/tabletraits.jl

```diff
@@ -0,0 +1,72 @@
+using TableTraits
+
+TableTraits.isiterable(x::IndexedTable) = true
+TableTraits.isiterabletable(x::IndexedTable) = true
+
+function TableTraits.getiterator{S<:IndexedTable}(source::S)
+    return rows(source)
+end
+
+# Sink
+
+@generated function _fillIndexedTable{idx_indices,data_indices}(iter,idx_storage,data_storage,::Type{idx_indices},::Type{data_indices})
+    push_exprs = Expr(:block)
+    for (i,idx) in enumerate(map(i->i.parameters[1],idx_indices.parameters))
+        ex = :( push!(idx_storage.columns[$i], row[$idx]) )
+        push!(push_exprs.args, ex)
+    end
+
+    for (i,idx) in enumerate(map(i->i.parameters[1],data_indices.parameters))
+        ex = :( push!(data_storage.columns[$i], row[$idx]) )
+        push!(push_exprs.args, ex)
+    end
+
+    quote
+        for row in iter
+            $push_exprs
+        end
+    end
+end
+
+function IndexedTable(x; idxcols::Union{Void,Vector{Symbol}}=nothing, datacols::Union{Void,Vector{Symbol}}=nothing)
+    if isiterabletable(x)
+        iter = getiterator(x)
+
+        source_colnames = TableTraits.column_names(iter)
+        source_coltypes = TableTraits.column_types(iter)
+
+        if idxcols==nothing && datacols==nothing
```

Comment:
Index columns need unique values, right? So one could also say that in this case it should create an ``IndexedTable`` without any index, so that this conversion works for any table data without having things like a unique requirement. But I'm not sure, I think the main thing is to be consistent across the different ways to create things both in ``IndexedTable`` and ``JuliaDB``.